src/dealDashboard/dealMenubar/dealMenubar.ts

```diff
@@ -19,9 +19,14 @@ export class DealMenubar {
     this.eventAggregator.publish("showMessage", "Deal URL copied");
   }
 
-  @computedFrom("deal.status")
+  @computedFrom("deal.status", "deal.isCancelled")
   get canEdit() {
-    return !this.deal.isExecuted && !this.deal.isCancelled;
+    return this.deal.isUserProposalLead && (!this.deal.isCancelled || !this.deal.isExecuted);
```

Comment:
this tells me that it can be edited even if is executed, which is not correct, or if is cancelled, which is not correct.